### PR TITLE
Contribute page

### LINKED ISSUE #1226: Create a "contribute" page to attact contributors
NUSMods is run entirely by students, for students. This means we're constantly looking for developers, designers and feedback, and now that we have venue information we also need students to contribute location info. 

Currently we have such a page on https://nusmods.com/about, but it's a bit lame. I think we can do better, and put the dedicated contribute button on the sidebar (on desktop) to make it more prominent. Ideas for what to put on the page to attract more contributors are welcome, but here are mine 

- The blurb from the current About page and the three buttons 
- A list of locations that are unmapped, and a progress bar for how many are left 
- The top/new contributors from the [contributors page](https://nusmods.com/contributors)
- Donation link as per #1120 

To build the page you can copy an existing static page from https://github.com/nusmodifications/nusmods/tree/master/www/src/js/views/static

### PR DESCRIPTION
**Pull request by monmanuela**
This is a WIP to create a 'Contribute' page to attract more contributors to NUSMods (Fixes #1226)

Changes made:
- [x] Add a dedicated 'Contribute' button on the sidebar
- [x] Put a short write-up
- [x] Show top contributors
- [x] Extract out common code for fetching of contributors to `js/apis`
- [x] Include Backer & Sponsor button
- [x] Configure progress bar to show how many unmapped venues are left
- [x] Show a list of unmapped venues
- [ ] Prettify buttons

If the refactoring of fetch contributors is ok, I can also modify the Contributors page to call the api instead.

### CODE REVIEW

File: www/src/js/apis/contributor.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// @flow
+import axios from 'axios';
+import type { Contributor } from 'types/contributor';
+
+const CONTRIBUTORS_URL =
+  'https://api.github.com/repos/NUSModifications/NUSMods/contributors?per_page=100';
```
Comment (yangshun) [inline www/src/js/apis/contributor.js]: Lower case URL please

File: www/src/js/views/components/UnmappedVenues.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
+// @flow
```
[the remark this answers is not in the document]
Comment (monmanuela) [reply www/src/js/views/components/UnmappedVenues.jsx]: To clarify, the `UnmappedVenues.jsx` should be in a subfolder like `views/unmapped-venues`?

File: www/src/js/views/components/UnmappedVenues.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import { fetchVenueList } from 'actions/venueBank';
+
+import venueLocations from 'data/venues.json';
+// import styles from './UnmappedVenues.scss';
+
+type Props = {
+  venueList: VenueList,
```
[the remark this answers is not in the document]
Comment (monmanuela) [reply www/src/js/views/components/UnmappedVenues.jsx]: My bad, seems like I forgot to stage the import line before committing

File: www/src/js/views/components/UnmappedVenues.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    this.props.fetchVenueList();
+  }
+
+  shouldComponentUpdate(nextProps: Props) {
+    return this.props.venueList.length !== nextProps.venueList.length;
+  }
+
+  render() {
+    const percentageMapped =
+      (Object.keys(venueLocations).length / this.props.venueList.length) * 100;
```
[the remark this answers is not in the document]
Comment (monmanuela) [reply www/src/js/views/components/UnmappedVenues.jsx]: Not sure at which component lifecycle should I calculate the `percentageMapped`. Initially, I thought of making `percentageMapped` a state and calculate it in `componentDidMount`, but I can't call `setState` in `componentDidMount`. Or perhaps I can leave it inside `render` and wrap it inside `{this.props.venueList && ...}`?

File: www/src/js/views/components/UnmappedVenues.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ZhangYiJiang) [reply www/src/js/views/components/UnmappedVenues.jsx]: No, you should create a directory `views/contribute` and put this and `ContributeContainer.jsx` in it.

File: www/src/js/views/components/UnmappedVenues.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (ZhangYiJiang) [reply www/src/js/views/components/UnmappedVenues.jsx]: You should not do this calculation at all if `venueLocations` is empty. Either return `null` (display nothing until it is loaded) or display a placeholder like the loading spinner

**Comment by ZhangYiJiang** [comment]
@monmanuela Do you have any updates?

**Comment by monmanuela** [comment]
@ZhangYiJiang Sorry, been out of town the past few weeks! Will work on it in the coming days

**Comment by ZhangYiJiang** [comment]
@monmanuela No problem, just checking if you have any commits not pushed, because I'll probably be cherry picking out the changes to the GitHub API by adding a script to pull the data on the server so that we don't get rate limit errors

**Comment by monmanuela** [comment]
@ZhangYiJiang ready for review. Not sure how to make the buttons more pretty so, for now, I leave it that way.

**Comment by ZhangYiJiang** [comment]
Fixed up most of the issues. I want to be able to link directly to the venue feedback form, but that's probably best for another PR. 

https://deploy-preview-1333--nusmods.netlify.com/contribute

- Fixed progress math 
- Venue is now loaded async via `Loadable.Map`
- Improve button design (still looks a bit ugly though) 
- Added some SVG icons to make the page prettier 

@taneliang  @li-kai code review  
@jiholim design review

File: www/src/js/apis/contributor.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+const CONTRIBUTOR_ID_RENOVATE = 25180681;
+
+export default function getContributors(): Promise<Contributor[]> {
+  return axios.get(CONTRIBUTORS_URL).then((response) =>
+    response.data.filter(
+      (contributor) =>
+        contributor.type === CONTRIBUTOR_TYPE_USER &&
+        // Renovate used to report outdated dependencies as a user via the GitHub API,
+        // hence we need to filter it out by its GitHub user ID.
+        contributor.id !== CONTRIBUTOR_ID_RENOVATE,
```
[the remark this answers is not in the document]
Comment (ZhangYiJiang) [reply www/src/js/apis/contributor.js]: This is being rewritten in #1353 anyway

**Pull request merged**